scripts/m18_submission_checker/aida_m18_submission_checker.py

```diff
@@ -0,0 +1,295 @@
+        return 'tar'
+    elif archive_file_name.endswith('.zip'):
+        return 'zip'
+    else:
+        logging.error("Archive filetype is unknown. Please use .zip, .tgz, or .tar.gz")
+        return None
+
+def get_archive_member_names(archive_file_name, archive_file_type):
+    """
+    Gets all of the file names/members of the archive.
+    :param archive_file_name: the file name/path of the archive
+    :param archive_file_type: the file extention type: 'tar' or 'zip'
+    :return: a list of the names/members in the archive on success, None if failure to open/read archive
+    :rtype: bool
+    """
+
+    logging.info("Checking archive... this may take a moment for large files.")
+
+    try:
+        archive = zipfile.ZipFile(archive_file_name, 'r', allowZip64=True) if archive_file_type == 'zip' else tarfile.open(archive_file_name, mode='r')
+        member_names_list = archive.namelist() if archive_file_type == 'zip' else archive.getnames()
+        return member_names_list
+    except:
+        # expected errors: (zipfile.BadZipFile, tarfile.ReadError), but catch all of them anyways
+        logging.error("Error thrown attempting to read/open the archive. Please use 'zip' or 'tar' to create your archive.")
+        return None
+
+def get_task_type(archive_file_name, archive_member_names):
+    """
+    :param archive_member_names: a list of members/names in the archive
+    :return: string corresponding to the task type, which could be any of the following: '1a', '1b', '2', or '3'
+    :rtype: string
+    """
+
+    # count the number of .'s in the archive file name to determine the task type. Hope the user has the proper naming convention
+    basename = os.path.basename(archive_file_name)
+
+    dot_count = basename.count('.')
+
+    # get rid of extra . count if the file extention is .tar.gz
+    if basename.endswith('.tar.gz'):
+        dot_count -= 1
+
+    if dot_count == 1:
+        # if theres only 1 dot_count, we need to check to see if it is 1a or 1b.
+
+        member_names_str = " ".join(archive_member_names)
+
+        # if .ttl file in "/NIST/" directory then ttls_under_nist_dir is not none (possible 1a)
+        ttls_under_nist_dir = re.compile(".*\/NIST\/[^\/]+.ttl", re.IGNORECASE).search(member_names_str)
+        # if ".ttl file in a subdirectory of /NIST/" then ttls_under_nist_subdir is not none (possible 1b)
+        ttls_under_nist_subdir = re.compile(".*\/NIST\/\S*\/\S*\.ttl", re.IGNORECASE).search(member_names_str)
+
+        if ttls_under_nist_dir is not None and ttls_under_nist_subdir is None:
+            return '1a'
+        elif ttls_under_nist_subdir is not None and ttls_under_nist_dir is None:
+            return '1b'
+        else:
+            # ttls found in /NIST/ and a subdirectory of /NIST/, so we cannot determine if 1a or 1b.
+            logging.warning("Cannot distinguish between 1a or 1b due to .ttl files existing at the same level as a subdirectory of /NIST/. \nContinuing as 1a.")
```

Comment:
next commit will not change the way it distinguishes between 1a and 1b, however it WILL error upon no ttl files in a NIST directory or no NIST directory.